src/app/public/src/modules/edit-button/edit-button.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { Router } from '@angular/router';
+
+import { StacheJsonDataService } from '../shared';
+import { StacheNavService } from '../nav';
+
+@Component({
+  selector: 'stache-edit-button',
+  templateUrl: './edit-button.component.html',
+  styleUrls: ['./edit-button.component.scss']
+})
+export class StacheEditButtonComponent implements OnInit {
+  @Input()
+  public editButtonText = 'Edit';
+
+  private url: string;
+  private routeFragment: string;
+
+  public constructor (
+    private dataService: StacheJsonDataService,
+    private navService: StacheNavService,
+    private router: Router
+  ) { }
+
+  public ngOnInit() {
+    this.routeFragment = this.router.url.split('#')[0].substring(1);
```

Comment:
if you pull in the `StacheRouteService` you can make use of the method we already have written
https://github.com/blackbaud/stache2/blob/master/src/app/public/src/modules/shared/route.service.ts#L53